pkg/cloud/azureprovider.go

```diff
@@ -923,6 +923,19 @@ func (az *Azure) DownloadPricingData() error {
 	return nil
 }
 
+// determineCloudByRegion uses region name to pick the correct Cloud Environment for the azure provider to use
+func determineCloudByRegion(region string) azure.Environment{
+	lcRegion := strings.ToLower(region)
+	if strings.Contains(lcRegion, "china") {
+		return azure.ChinaCloud
+	}
+	if strings.Contains(lcRegion, "gov") || strings.Contains(lcRegion, "dod") {
+		return azure.USGovernmentCloud
+	}
```

Comment:
Is there any possibility of `region` containing one of these substrings but not being what we expect? I'm not sure about the possible options for `region` -- maybe we could do `strings.StartsWith` if all China regions start with `china` for example. I can't seem to find a full list of Azure regions that includes China and govcloud, so I'm not sure how to evaluate this.

Reply:
I could do starts with there, would be a little more robust.
This is the link you are looking for https://azure.microsoft.com/en-us/global-infrastructure/geographies/#geographies

Reply:
and yeah was a little concerned about "dod" and "gov" but there are no collisions there